projects/packages/sync/src/modules/class-meta.php

```diff
@@ -48,13 +50,66 @@ public function get_objects_by_id( $object_type, $config ) {
 			return array();
 		}
 
-		$meta_objects = array();
+		$object_id_column = $object_type . '_id';
+		$object_key_pairs = array();
+
 		foreach ( $config as $item ) {
-			$meta = null;
 			if ( isset( $item['id'] ) && isset( $item['meta_key'] ) ) {
-				$meta = $this->get_object_by_id( $object_type, (int) $item['id'], (string) $item['meta_key'] );
+				$object_key_pairs[ (int) $item['id'] ][] = (string) $item['meta_key'];
+			}
+		}
+
+		$conditionals         = array();
+		$where_sql            = '';
+		$current_query_length = 0;
+
+		foreach ( $object_key_pairs as $object_id => $keys ) {
```

Comment:
Is there a reason to have a new foreach instead of having everything done with the one in line 56?

Reply:
We are grouping by `object_id` first to reduce the query length essentially.